examples/Clients/Counter/Program.cs

```diff
@@ -34,7 +36,8 @@ static async Task Main(string[] args)
             // Server will only support Https on Windows and Linux
             var credentials = RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? ChannelCredentials.Insecure : ClientResources.SslCredentials;
             var channel = new Channel("localhost:50051", credentials);
-            var client = new Counter.CounterClient(channel);
+            //var client = new Counter.CounterClient(channel);
+            var client = GrpcClientFactory.Create<Counter.CounterClient>("https://localhost:50051", new System.Security.Cryptography.X509Certificates.X509Certificate2(Path.Combine(Resources.CertDir, "client.crt")));
```

Comment:
Ignore changes in the samples. They are me playing around. I'll clean them up when this is closer to done.